zarf.schema.json

```diff
@@ -1003,7 +1003,7 @@
       "properties": {
         "name": {
           "type": "string",
-          "pattern": "^[a-z0-9\\-]*[a-z0-9]$",
+          "pattern": "^[a-z0-9][a-z0-9\\-]*$",
```

Comment:
What drove this change?

Reply:
It's the wrong regex, we currently do a check here https://github.com/defenseunicorns/zarf/blob/main/src/types/validate.go#L26 at create time. I'm going to refactor to use the schema instead of a separate test, but wanted to get the schema in line and create tests first

Reply:
Dope :)